rs/execution_environment/src/canister_manager/tests.rs

```diff
@@ -544,219 +478,6 @@ fn upgrade_canister_with_no_wasm_fails() {
-    );
-    assert_eq!(
-        test.canister_state(canister_id3).compute_allocation(),
-        ComputeAllocation::try_from(20).unwrap()
-    );
-}
-
-#[test]
-fn install_canister_fails_if_memory_capacity_exceeded() {
-    let initial_cycles = Cycles::new(1_000_000_000_000_000);
-    let mb = 1 << 20;
-    // canister history memory usage for canister1 at the beginning of install_code
-    let canister_history_memory_usage = size_of::<CanisterChange>() + size_of::<PrincipalId>();
-    let memory_capacity = 1000 * mb;
-    // canister1 memory usage before code change: `canister_history_memory_usage`
-    // canister1 memory usage after code change: `memory_used`
-    // => SubnetAvailableMemory decreases by `memory_used - canister_history_memory_usage`
-    // after canister1 code change and then SubnetAvailableMemory is equal to
-    // `memory_capacity - (memory_used - canister_history_memory_usage)`;
-    // we want this quantity to be `10 * mb` and derive the value of `memory_used` from there.
-    let memory_used = memory_capacity - 10 * mb + (canister_history_memory_usage as u64);
-
-    let wat = r#"
-        (module
-            (func (export "canister_init")
-                (drop (memory.grow (i32.const 160)))
-            )
-            (memory 0)
-        )"#;
-
-    let mut test = ExecutionTestBuilder::new()
-        .with_subnet_execution_memory(memory_capacity as i64)
-        .with_subnet_memory_reservation(0)
-        .build();
-
-    let wasm = wat::parse_str(wat).unwrap();
-
-    let canister1 = test.create_canister(initial_cycles);
-    let canister2 = test.create_canister(initial_cycles);
-
-    test.install_canister_with_allocation(canister1, wasm.clone(), None, Some(memory_used))
-        .unwrap();
-
-    let execution_cost_before = test.canister_execution_cost(canister2);
-    let err = test
-        .install_canister_with_allocation(canister2, wasm.clone(), None, Some(11 * mb))
-        .unwrap_err();
-
-    err.assert_contains(
-        ErrorCode::SubnetOversubscribed,
-        "Canister requested 11.00 MiB of memory but only 10.00 MiB are available in the subnet.",
-    );
-    // The memory allocation is validated first before charging the fee.
-    assert_eq!(
-        test.canister_state(canister2).system_state.balance(),
-        initial_cycles - (test.canister_execution_cost(canister2) - execution_cost_before),
-    );
-
-    // Try installing without any memory allocation.
-    let err = test
-        .install_canister_with_allocation(canister2, wasm, None, None)
-        .unwrap_err();
-    let execution_cost_after = test.canister_execution_cost(canister2);
-    err.assert_contains(
-        ErrorCode::SubnetOversubscribed,
-        "Canister requested 10.00 MiB of memory but only 10.00 MiB are available in the subnet.",
-    );
-
-    assert_eq!(
-        test.canister_state(canister2).system_state.balance(),
-        initial_cycles - (execution_cost_after - execution_cost_before)
-    );
```

Comment:
Is this covered elsewhere?

Reply:
Good catch. I should keep the cases where we test without memory allocation.